nurseconnect/forms.py

```diff
@@ -136,22 +142,36 @@ def security_questions(self):
             )
         ]
 
-    def clean_username(self):
-        if User.objects.filter(
-            username__iexact=self.cleaned_data["username"]
-        ).exists():
-            raise forms.ValidationError(_("Username already exists."))
 
-        return self.cleaned_data["username"]
+class RegistrationClinicCodeForm(forms.Form):
+    clinic_code = forms.RegexField(
+        regex=r"^\d{6}$",
+        required=True,
+        label=_("Clinic code"),
+        error_messages={
+            "invalid": "Please enter your 6 digit clinic code"
+        },
+        widget=forms.TextInput(
+            attrs={
+                "placeholder": _("Clinic code"),
+                "class": "Form-input",
+                "for": "cliniccode"
+            }
+        ),
+    )
 
-    def clean(self):
-        password = self.cleaned_data.get("password", None)
-        confirm_password = self.cleaned_data.get("confirm_password", None)
-        if (password and confirm_password and
-                (password == confirm_password)):
-            return self.cleaned_data
+    def clean_clinic_code(self):
+        clinic_code = self.cleaned_data["clinic_code"]
+
+        clinic = check_clinic_code(clinic_code)
```

Comment:
Having this function here tightly couples the form to that API service. Isn't there a way we can move this to a corresponding view instead?